corporate/views/upgrade.py

```diff
@@ -231,6 +231,12 @@ def sponsorship(
     # an underscore.
     post_data.update(organization_type=organization_type)
     form = SponsorshipRequestForm(post_data)
+    # We do this so that even if `form.is_valid()` returns False
+    # we can still try to determine a display name for the given
+    # org type and send it in the email below. Other parts of the
+    # form being invalid does not necessarily mean that the org
+    # type is invalid.
```

Comment:
Shouldn't we just reject the request if the form is invalid?  

The architecture of doing some but not all things with an invalid form submission doesn't make sense to me, and I think cleaning that up will fix this as well.

Reply:
We could reject yes. My rationale for this was that if you look at the alert where we caught this, the sponsorship request was from a person who provided an invalid URL but everything else was valid. I guess the question is, do we still wanna review sponsorship requests that don't have a URL, for instance?

Reply:
We could also just make the URL validation less stringent and use a char field to allow people without websites to still submit sponsorship requests, I guess.

Reply:
I think it might be better to just have a "Organization doesn't have a URL" checkbox, which submits `""` for the URL, and turn that into a `None` for the field in the database (without validating).